lib/molinillo/resolution.rb

```diff
@@ -230,8 +239,13 @@ def state_index_for_unwind
         index = -1
         maximal_index = states.size - 2
         conflicts.each do |dependency_name, conflict|
-          next unless activated.vertex_named(dependency_name)
-          conflict.requirements.values.flatten(1).uniq.each do |r|
+          requirements_to_attempt_to_relax = [
+            conflict.requirement,
+            initial_requirement_for_name(dependency_name),
+            *binding_requirements_for_conflict(conflict),
+          ].compact.uniq
+
+          requirements_to_attempt_to_relax.each do |r|
```

Comment:
It's a big speedup - cut spec run from 45 seconds to 11 on my machine. Cuts out cases where we unwind less than we could have done - if we unwind to a non-binding requirement that still has possibilities, we can sometimes do another 10 or so steps before getting back to the same fundamental conflict. I'll add a comment.